fix issue #810

xlla opened the pull request:
to solve the error
<pre>
[rviz2-3] [ERROR] [1652204061.476182910] [rviz2]: ItemIdentityException: Can not find codec for 'tif' image format.
[rviz2-3] Supported formats are: bmp dds gif hdr jpeg jpg ktx pgm pic pkm png ppm psd tga. in Codec::getCodec at /Users/xlla/ros2_foxy/build/rviz_ogre_vendor/ogre-v1.12.1-prefix/src/ogre-v1.12.1/OgreMain/src/OgreCodec.cpp (line 66)
</pre>

xlla committed:
fix issue #810

Signed-off-by: xlla <xllacyx@hotmail.com>

clalancette reviewed (changes requested):
First of all, thanks for starting this.

Unfortunately, this change on its own is not enough.  In particular, we also need to make sure that libfreeimage-dev is installed on the system before we use this, otherwise we get errors like:

```
Could not load dynamic library install/rviz_ogre_vendor/opt/rviz_ogre_vendor/lib/OGRE/Codec_FreeImage
```

To do that, we need to add `<depend>libfreeimage-dev</depend>` to `rviz_ogre_vendor`, and we also need to add `libfreeimage` to the rosdep database.  I've opened a pull request for the latter at https://github.com/ros/rosdistro/pull/33403 .

We also need to test this out on Windows to make sure that it works there.  Finally, we should target this to the `ros2` branch first, as we want to get this onto Rolling, and then think about backporting to Humble.

clalancette commented:
> To do that, we need to add `<depend>libfreeimage-dev</depend>` to `rviz_ogre_vendor`, and we also need to add `libfreeimage` to the rosdep database.  I've opened a pull request for the latter at [ros/rosdistro#33403](https://github.com/ros/rosdistro/pull/33403) .

All right, that PR is in.  I'm going to re-target this to the `ros2` branch, and also add those rosdep keys to this PR.  That should solve the problem for Linux.

For Windows, I think we'll have to add `freeimage` as part of the build process in `rviz_ogre_vendor`.  If we do that, that should make this work on all platforms.

clalancette committed:
Add libfreeimage dependencies to rviz_ogre_vendor.

That way freeimage is available and OGRE support for it is
built when building this vendor package.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>
